packages/components/src/custom-select-control/test/index.js

```diff
@@ -87,4 +88,127 @@ describe( 'CustomSelectControl', () => {
 			screen.getByRole( 'button', { name: 'Custom select' } )
 		).toHaveTextContent( 'Hint' );
 	} );
+
+	it( 'Initial value should be replaced when a new item is selected', async () => {
+		const user = userEvent.setup();
+		const mock = jest.fn();
+
+		render(
+			<CustomSelectControl
+				onChange={ ( { selectedItem } ) => mock( selectedItem ) }
+				options={ options }
+				__nextUnconstrainedWidth
+			/>
+		);
+
+		const currentValue = screen.getByRole( 'button' );
+
+		expect( currentValue ).toHaveTextContent( /violets/i );
+
+		await user.click( currentValue );
+
+		const newValue = screen.getByRole( 'option', {
+			name: /poppy/i,
+		} );
+
+		await user.click( newValue );
+
+		expect( currentValue ).toHaveTextContent( /poppy/i );
+		expect( mock ).toHaveBeenCalled();
+	} );
+
+	it( 'Should be able to change selection using keyboard', async () => {
+		const user = userEvent.setup();
+
+		render(
+			<CustomSelectControl options={ options } __nextUnconstrainedWidth />
+		);
+
+		const currentValue = screen.getByRole( 'button' );
+		expect( currentValue ).toHaveTextContent( /violets/i );
+
+		await user.tab();
```

Comment:
Tests that rely on the fact that an element is the first one in order can be fragile, because they'll break as soon as we change the order of elements. Would it be less fragile if we click on the `currentValue`?

Reply:
The intention of this specific test is to test that the keyboard functionality works as expected, so I think it might be worth keeping the tab here. This is why there is also an assertion to check the focus. 

But there is another test that could use this update. 👍